### ISSUE TITLE
Sigils with capital letter not accepted by FunctionNames

### ISSUE THREAD

**Issue by lasseebert** [rrrene/credo#679]
### Environment

* Credo version (`mix credo -v`): 1.1.1
* Erlang/Elixir version (`elixir -v`): Elixir 1.9.1 (compiled with Erlang/OTP 22) (Erlang 22.0.7)
* Operating system: Ubuntu

### What were you trying to do?

Defining a custom sigil with a capital letter (for example like the built in `~S`):

```elixir
def sigil_O(input, args) do
  # ...
end
```

### Expected outcome

Since this is the Elixir way of defining capital letter sigils, it should not warn.

### Actual outcome

FunctionNames check warns:

```
  Code Readability                                                              
┃ 
┃ [R] ↗ Function/macro/guard names should be written in snake_case.
┃       apps/butterbur/lib/butterbur/calculation/ratio.ex:25:7 #(Butterbur.Ratio.sigil_O)
┃ [R] ↗ Function/macro/guard names should be written in snake_case.
┃       apps/butterbur/lib/butterbur/calculation/quantity.ex:19:7 #(Butterbur.Quantity.sigil_Q)
```

This issue seems to have been introduced in credo 1.1.1 and was not present in 1.1.0.

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by lasseebert** [comment]
@rrrene Thanks a lot for the quick fix :)
I have tested on master, and it does not report a warning now :+1:

**Comment by rrrene** [comment]
@lasseebert The fix is now also part of `v1.1.2` :+1: Thanks again for reporting this!
